### PR TITLE
Add sign operation

### PR DESCRIPTION
**Pull request by wyuinche**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feat : add sign operation

* **What is the current behavior?** (You can also link to an open issue here)
1. No sign operation for creators and copyrighters of nft.
2. creator/copyrighter named right-holder.

* **What is the new behavior (if this is a feature change)?**
1. Now, a user can sign as a creator or copyrighter of the nft.
2. right-holder has been renamed signer.

* **Other information**:

### CODE REVIEW

File: nft/nft.go [hunk 1]
```diff
@@ -35,11 +35,11 @@ type NFT struct {
 	hash         NFTHash
 	uri          URI
 	approved     base.Address
-	creators     []RightHolder
-	copyrighters []RightHolder
+	creators     []Signer
```
Comment (soonkuk) [inline nft/nft.go:38]: @wyuinche 
The number of signers and creators also needs to be limited.

**Review by soonkuk — CHANGES_REQUESTED** [review]

**Review by soonkuk — CHANGES_REQUESTED** [review]

File: nft/collection/sign_process.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+	// check owner
+	if n.Owner().String() == "" {
+		return errors.Errorf("dead nft; %q", nid)
+	}
+
+	var idx = -1
+	for i := range holders {
+		if holders[i].Account().Equal(ipp.sender) {
+			idx = i
```
Comment (soonkuk) [inline nft/collection/sign_process.go:93]: @wyuinche 
Why is the use of idx necessary? I think it is enough to just return an error right inside the for statement.

**Review by soonkuk — CHANGES_REQUESTED** [review]

File: nft/nft.go (hunk 1 above)
Comment (soonkuk) [reply nft/nft.go:43]: @wyuinche
Is there any validation check for duplicated address in creators and duplicated address in copywrighters?

Comment (wyuinche) [reply nft/nft.go:43]: @soonkuk
A limit has been applied to creators and copyrighters. The current limit is 10.

Comment (wyuinche) [reply nft/nft.go:43]: > @wyuinche Is there any validation check for duplicated address in creators and duplicated address in copywrighters?

@soonkuk No🥲 I'll address this problem.

Comment (wyuinche) [reply nft/nft.go:43]: > > @wyuinche Is there any validation check for duplicated address in creators and duplicated address in copywrighters?
> 
> @soonkuk No🥲 I'll address this problem.

@soonkuk 
Added duplicate detection for signers of 'nft', 'mint-form'.

File: nft/nft.go [hunk 3]
```diff
@@ -115,16 +124,32 @@ func (n NFT) IsValid([]byte) error {
 		return isvalid.InvalidError.Errorf("copyrighters over allowed; %d > %d", l, MaxCopyrighters)
 	}
 
+	foundSigner := map[base.Address]bool{}
```
Comment (soonkuk) [inline nft/nft.go:127]: It is more recommended
`foundSigner := map[base.Address]struct{}{}`
because `struct{}` consumes no storage.

**Review by soonkuk — CHANGES_REQUESTED** [review]

File: nft/nft.go (hunk 3 above)
Comment (wyuinche) [reply nft/nft.go]: @soonkuk I've addressed it!

**Review by soonkuk — APPROVED** [review]
LGTM!

**Pull request merged**